### ISSUE TITLE
How could we pass mock/test data into components?

### ISSUE THREAD

**Issue by bryantAXS** [trendyminds/molecule#3]
This is a tangentially related issue to Molecule, but I'm wondering what the best way would be to organize mock data and load that into the component?

Ideally, it would be great if every component was able to be mocked with static data not retrieved from the CMS, and in a way where we're not just setting default data fallbacks inside the component itself.

I'm wondering if something like creating a `mock.twig` file inside the component directory and then somehow loading data set in there into the component would be possible?

```
├── components/
│   ├── ButtonPrimary/
│   │   ├── index.css
│   │   ├── index.jsx
│   │   ├── index.twig
│   │   ├── mock.twig
```

Sorry for such a vague question, I'm sort of trying to wrap my head around the problem still :)

By the way, great work here! I love Molecule and the way you're thinking about components and templating.

**Comment by bryantAXS** [comment]
@aaronbushnell I added a potential update here to get your thoughts.

https://github.com/bryantAXS/molecule/commit/27132b1e8e753ff645012cd4c7b219488df61ff8

**Comment by aaronbushnell** [comment]
Hey @bryantAXS do you mind passing along a use-case or two on how something like this would work? Thanks for sending over the commit!

**Comment by bryantAXS** [comment]
@aaronbushnell So the end goal is to make components as portable and flexible as possible and be able to develop them without the need for the CMS (or potentially before the CMS data has been configured)

On the project we're currently working on, I created this "component viewer" page to let us development components without the need of a CMS. https://cl.ly/bbe206d011a1 A big part of this workflow is being able to load in data using the approach I added, so that we're not just hardcoding in dummy data inside a single file for all the components.

Being able to keep mock data inside the component folder system is just tidier and easier to manage, and I think follows the same line of component-driven thinking that you're going for with Molecule. My approach could surely be improved, but the biggest thing is just a way to specify data independent of the database, and in a way that's easy to manage.

Does that make sense?
